File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 7/28/2025, 12:51:28 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int min_n = INT_MAX, profit = 0;

        for(int i = 0; i<n; i++){
            min_n = min(min_n, prices[i]);
            profit = max(profit, prices[i] - min_n);
        }
        
        return profit;
    }
};

File: 121-BestTimetoBuyandSellStock/121-BestTimetoBuyandSellStock.py
# Last updated: 8/4/2025, 9:47:13 AM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy, sell = 0, 1
        prof = 0

        while sell < len(prices):
            if prices[buy] < prices[sell] :
                m = prices[sell] - prices[buy]
                prof = max(m, prof)
            else :
                buy = sell
            sell += 1
        return prof